docusaurus-search-local/src/client/theme/SearchBar/SearchBar.tsx

```diff
@@ -231,6 +235,16 @@ export default function SearchBar({
     };
   }, [isMac, onInputFocus]);
 
+  const onClearSearch = useCallback(() => {
+    const params = new URLSearchParams(window.location.search);
+    params.delete(SEARCH_PARAM_HIGHLIGHT);
+    history.push(
+      window.location.pathname + `?${params.toString()}` + window.location.hash
```

Comment:
We should use `location` from `useLocation` of `@docusaurus/router`:

```suggestion
      location.pathname + `?${params.toString()}` + location.hash
```

Reply:
I'm not sure why but the location is not correct here, for example if you're on a doc page, lets say `/docs/mdx?_highlight=jsx` then location.pathname is `/docs/` and hash is empty.

Reply:
Put the `[location.pathname, location.search, location.hash]` into the dependency argument of `useCallback`?

Reply:
I think there will be problems if just using `window.location` while `baseUrl` is set to a subdirectory.

Reply:
It actually works fine for that case too. Since we just want to reload the exact url we're on. But yes I can try that, would be nicer if it worked with location. I need to read up on how those work.

Reply:
Using the dependency list was ofc the solution, should be fixed now, thanks :)